### ISSUE TITLE
"inappstore" vulnerability. Please upgrade it from Store Kit 1 to Store Kit 2

### ISSUE THREAD

**Issue by PickleJesus123** [godotengine/godot-ios-plugins#57]
There is a vulnerability with the current "inappstore" plugin.

Unlike the Google Play Billing plugin, there is no way to `query_purchases()` and find out what the user has purchased/subscribed to when the app starts up. This is due to the plugin using StoreKit 1. Apple's StoreKit 2, however, provides functionality out of the box that allows the user to query all transactions.

I assume most Godot game developers using this plugin are just using the following logic:
```
func check_events():
	while in_app_store.get_pending_event_count() > 0:
		var event = in_app_store.pop_pending_event()
		if event.type == "purchase":
			if event.result == "ok":
				// save a local variable somewhere that "purchased = true"
```

Some notes about this:
- For consumable purchases, this is no problem
- For "permanent" purchases, this should mostly work. But if a malicious player buys and then quickly refunds your item, you would have no way of telling. There is no "event fired" in StoreKit 1. To detect refunds, you would need to [maintain your own server](https://developer.apple.com/documentation/appstoreserverapi) that is constantly in contact with Apple's servers, and your app would have to contact your server to revoke the premium features. I guarantee this is out of scope for most Godot game developers.
- For "auto-renewing subscriptions", simply being able to read basic information like `expiry_date` involves maintaining your own server that can do [receiptValidation](https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/validating_receipts_with_the_app_store). If you want it to detect refunds, renewals, or cancellations, you need to implement the whole App Store Server Api on your server.
- TLDR: unless you are maintaining your own complex server infrastructure, your Godot game is going to be vulnerable to purchase-refund scams, and modding (setting purchased = true). I doubt many Godot game developers are.

The new StoreKit 2 is designed to simplify this whole process. You can just call `query_purchases()` and get a list of active purchases - that's it! StoreKit 2 requires iOS 15+, but it is installed on **99%+ of iOS devices**. No compatibility concerns there. StoreKit 1 is a mess!

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-ios-plugins/issues/17.

**Comment by PickleJesus123** [comment]
I am the primary earner for my family, and I make 100% of my income from my mobile games. All of my games are made in Unity, but I have been in the process of converting all my games over to Godot ever since the "20 cents per install" fiasco. I almost finished porting my first game over to Godot. When it finally came time to add the "monetization features" to my game (which is how I earn **any** money from my games) everything came to a screeching halt.
- The Android Billing plugins had no active maintainers, and were so outdated that they technically became deprecated on Wednesday. I had to spam posts all over various Godot communities to even get a maintainer to "hack a temporary workaround" on the final day of deprecation. Even with the workaround, the plugin is still deprecated, and there are still no active maintainers.
- The iOS plugins don't have any contributors, so the plugins are 'rotting'.
- There is [only one](https://github.com/Poing-Studios/godot-admob-android) ad monetization plugin to choose from, maintained by a single dude. I don't like the idea of that.

Constructive criticism: What I've come to realize is that Godot is great for devs selling games on Steam, but for anyone who makes their income with mobile games, Godot is essentially a non-option. The features that mobile devs need are totally neglected and rotting, and have been for a while. These plugins need to be higher priority for Godot's development team, if they want mobile devs to use Godot. Maybe the Godot team doesn't care and just wants people to make Steam games - and that's fine, I must have gotten the wrong impression. The development fund is at €50,000+ per month, where are all the contributors?

**Comment by dinopanagos** [comment]
This has been lacking contributors and is pushing people away from Godot. Is there any way we can all chip in an just pay someone to take care of it. Been over 6 months with no updates or timeline. Godot is amazing and its just going to fade if independent devs cant monetize.

**Comment by Calinou** [comment]
> Is there any way we can all chip in an just pay someone to take care of it.

While you are free to contact a developer and pay them (including through a crowdfunding campaign you host), we don't use a bounty system. This is something you will have to organize on your own, including finding that developer and ensuring they do their work correctly. There's a lot of risk involved in this process by design.

**Comment by djordjije** [comment]
I am more than willing to work on the iOS plugins if I can successfully find some information on how to code them. Just give me some time.

**Comment by dinopanagos** [comment]
almost finished with it ! no worries :)

**Comment by djrain** [comment]
@dinopanagos hey, did you make any progress on this? If so, would you mind sharing the files?

**Comment by hayderkharrufa** [comment]
See:
https://github.com/rktprof/godot-ios-extensions

**Comment by omur-can-altundasar** [comment]
InAppStore plugin should be updated to Storekit 2. It's not good for security and Apple may not support it soon. It would be great if you update it please

**Comment by RocketRonz** [comment]
Mobile development seems to be pretty low priority for Godot, which is extremely frustrating. There is another plugin available but I haven't had a chance to try it in my game yet:

https://github.com/skyelynwaddell/godot-ios-plugin

**Comment by ivanfemia** [comment]
I use this plugin in my two mobile games 

https://github.com/zt-pawer/SwiftGodotIosPlugins

**Comment by BraveEvidence** [comment]
This might help https://www.youtube.com/watch?v=aRWMmONfbrA
